Integrations/integration-ATD.yml

```diff
@@ -37,672 +38,670 @@ configuration:
+        ''' Converts a string array to int array '''
+        intArr = []
+        if isinstance(stringArr, list):
+            for index in range(len(stringArr)):
+                if stringArr[index]:
+                    intArr.append(int(stringArr[index]))
+        else:
+            intArr = [int(stringArr)]
+        return intArr
+
+    def cast_to_int(val):
+        ''' Casts a string to int '''
+        if isinstance(val, str):
+            return int(val)
+        else:
+            return val
+
+    def get_session_credentials():
+        res = http_request('php/session.php','get', LOGINHEADERS)
+        if (res):
+            return res['results']
+
+    def get_headers():
+        sess = get_session_credentials()
+        return {
+            'Accept': 'application/vnd.ve.v1.0+json',
+            'VE-SDK-API': base64.b64encode(sess['session'] + ':' + sess['userId'])
         }
-    }
 
-    function getHeaders(){
-        var sess = getSessionCredentials();
-        return {'Accept': ['application/vnd.ve.v1.0+json'],
-                'VE-SDK-API':[Base64.encode(sess.session + ':' + sess.userId)]
-        };
-    }
-
-    function sendRequestRaw(uri, method, headers, body){
-        var res = http(
-                BASE+uri,
-                {
-                    Method: method,
-                    Headers: headers,
-                    Body: body
-                },
-                UNSECURE,
-                PROXY
-            );
-        if (res.StatusCode < 200 || res.StatusCode >= 300) {
-            throw 'Failed to use uri ' + BASE+uri + ', request status code: ' + res.StatusCode + ' and Body: ' + res.Body + '.';
-        }
-        var uriStr = String(uri);
-        if (uriStr.indexOf("session") === -1) {
-            logout();
+    def http_request(uri, method, headers = {}, body = {}, params={}, files={}):
+        ''' Makes an API call with the supplied uri, method, headers, body '''
+        url = '%s/%s' % (BASE_URL, uri)
+        res = requests.request(
+            method,
+            url,
+            headers=headers,
+            data=body,
+            verify=USE_SSL,
+            params=params,
+            files=files
+        )
+        if res.status_code < 200 or res.status_code >= 300:
+                return_error('Got status code ' + str(res.status_code) + ' with body ' + res.content + ' with headers ' + str(res.headers))
+        result = res.content
+        if not uri.startswith('php/showreport.php?'):
+            result = json.loads(result)
+            if 'success' in result:
+                if (result['success'] == 'false'):
+                    return_error('ATD Api call to ' + uri + ' failed with request body: ' + body)
+        return result
+
+
+
+    ''' FUNCTIONS '''
+    def heart_beat():
+        return http_request('php/heartbeat.php','get', API_HEADERS, HEARTBEATHEADERS)
+
+    def test_get_session():
+        res = http_request('php/session.php','get', LOGINHEADERS)
+
```

Comment:
Shouldn't be in helper functions section?

Reply:
get_session is the 'atd-login' command, so no